### PR TITLE
added handling for browser suppression of audio autoplay

### LINKED ISSUE #21: Audio doesn't playback on Safari unless getUserMedia is called
Safari requires explicit user consent to play audio on both Mac and iOS. Because our audio elements are created without explicit user interaction, they would not playback unless the user is also publishing their tracks. 

References:

* https://webkit.org/blog/7734/auto-play-policy-changes-for-macos/
* https://webkit.org/blog/6784/new-video-policies-for-ios/

### PR DESCRIPTION
**Pull request by davidzhao**
See README for details, fixes #21

### CODE REVIEW

**Review by frostbyte73 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -107,6 +107,30 @@ videoTrack.mute();
 room.localParticipant.unpublishTrack(videoTrack);
 ```
 
+### Audio playback
+
+Browsers can be restrictive regarding if audio could be played without user interaction. What each browser considers as user interaction can also be different (with Safari on iOS being the most restrictive). Some browser considers clicking on a button unrelated to audio as interaction, others require audio element's `play` function to be triggered by a onclick event.
```
Comment (OmriSama) [inline README.md:112]: Would rephrase:

> Browsers can be restrictive with regards to audio playback that is not initiated by some user interaction.
> What each browser considers a 'user interaction' can vary by vendor (with Safari on iOS being the most restrictive).
> Some browsers consider clicking on a button unrelated to audio as interaction, others require an `<audio>` element's `play` function to be triggered by an `onClick` event.

Comment (hn8) [reply README.md:112]: Actually even the strictest Webkit doesn't require user gesture in play if the media element is created in user gesture first.
https://github.com/WebKit/WebKit/blob/main/Source/WebCore/html/HTMLMediaElement.cpp#L440

Also user activation could even propogate through postMessage or setTimeout and not just limited to onclick
https://whatpr.org/html/3851/interaction.html#tracking-user-activation

But it is inconsistent across browsers, Chrome only whitelists 4 gestures
https://github.com/chromium/chromium/blob/master/content/browser/web_contents/web_contents_impl.cc#L239

Comment (davidzhao) [reply README.md:112]: thanks all for the suggestions. updated.
